operations/observability/Observability.md

```diff
@@ -0,0 +1,318 @@
+- There is a graph showing the duration for each stage and operation is added.
+
+![IUF timing dashboard](../../img/operations/TimingDashboard.png "Timing Dashboard")
+
+## Goss tests for PIT and NCN
+
+### Goss test overview
+
+The observability tooling monitors the logs for Goss tests run by using the automated scripts. These scripts are regularly run during install/upgrade. For each Goss test, metrics are generated on its duration and success or failure.
+
+This framework provides a set of quantifiable metrics used to create a visual Grafana health dashboard of all environments. This provides administrators with insight on which tests fail most, or which clusters have the most problems.
+
+Trend analysis of this data in Grafana across different dimensions may point out statistically where the highest frequency of issues occurs.
+It also provides administrators an at-a-glance dashboard where they can visually see the complete system health. This includes the option to drill down from the aggregate view of the environment to the functional areas or the components.
+
+Goss test logs are scraped using grok-exporter and visualized on Grafana using captured data.
+[`csm-testing` repository](https://github.com/Cray-HPE/csm-testing/tree/main/goss-testing/) contains all of the tests, suites, and scripts to execute these Goss tests.
+Automated scripts run the Goss tests in batches. These batches are based on functionality or the check performed in CSM.
+
+A single Goss test is a YAML file and a collection of these Goss tests can be used together by adding them in another YAML file, called a test suite. These suites can be invoked through the CSM-provided scripts to get a log file with results for the tests.
+
+For example, a Goss test to validate expected Kubernetes nodes exists using the `kubectl` command based on node names pulled from `/etc/hosts` file. The log files generated from automated scripts that run a set of test suites are used.
+
+In order to run the automated scripts, use the complete path of the script. For example, `/opt/cray/tests/install/ncn/automated/ncn-healthcheck`.
+
+### Workflow
+
+![Goss test workflow](../../img/operations/GossWorkflow.png "Goss Workflow")
+
+### Log file format
+
+Individual lines of logs are in the following format for each node or PIT node, test name, and source:
+
+```json
+{"log_timestamp": "20230118_094205.821955", "Product": "CSM", "log_script": "print_goss_json_results.py", "log_message": "Test result", "Description": "Validates that 'cray --version' is available and executes successfully on the local system.", "Execution Time (nanoseconds)": 1312368478, "Execution Time (seconds)": 1.312368478, "Node": "ncn-m001", "Result Code": 0, "Result String": "PASS", "Source": "http://ncn-m001.hmn:8997/ncn-healthcheck-master", "Test Name": "Command 'cray --version' Available", "Test Summary": "Command: command_available_1_cray_version: exit-status: matches expectation: [0]"}
```

Comment:
The changes I had added were reverted again, this is one-line again.

Reply:
I've cherry-picked my old commit back, please `git fetch && git pull` before making more changes to prevent these from getting lost.

Reply:
sure, performed git fetch && git pull